src/components/organisms/TriggersFormItems/ActionFormItems.tsx

```diff
@@ -0,0 +1,64 @@
+import {useState} from 'react';
+import {useFirstMountState} from 'react-use';
+
+import {Form, Select, Space} from 'antd';
+
+import {LabelsSelect} from '@molecules';
+
+import {useStore} from '@store';
+
+import {required} from '@utils/form';
+
+import ResourceTriggerSelect from './ResourceTriggerSelect';
+import TriggerSelectorSwitcher from './TriggerSelectorSwitcher';
+
+const ActionFormItems = () => {
+  const {triggersKeyMap} = useStore(state => ({
+    triggersKeyMap: state.triggersKeyMap!,
+  }));
+
+  const [switcherValue, setSwitcherValue] = useState<string | number>('label');
+
+  const isFirst = useFirstMountState();
+
+  const actionOptions = triggersKeyMap?.actions
+    .map((actionItem: string) =>
+      triggersKeyMap.executions.map(executionItem => {
+        const label = `${actionItem} ${executionItem}`;
+        return {label, value: label};
+      })
+    )
+    .flat();
+  return (
+    <>
+      <Form.Item label="Testkube action" required name="action" rules={[required]}>
+        <Select options={actionOptions} placeholder="Select a testkube related action" />
+      </Form.Item>
+      <Form.Item noStyle shouldUpdate>
+        {({getFieldValue}) => {
+          const label = getFieldValue('testLabelSelector');
+          const name = getFieldValue('testNameSelector');
+
+          if (isFirst) {
+            setSwitcherValue(name ? 'name' : 'label');
+          }
```

Comment:
* Calling `setState` during render causes errors
* Should it really happen on the first mount only? If so, probably `useEffect(() => {}, [])` could be used

Reply:
- What kind of errors in this case?
- Do you think it is clear to extract `name` and pass it as props so I can check it is `useEffect`? It adds too much complexity I think

Reply:
React internal errors, like:
```
Cannot update during an existing state transition (such as within render or another component's constructor). Render methods should be a pure function of props and state.
```

Reply:
On the other side, when digging more in the code, I think that the whole idea of `ActionFormItems` has a flaw - it's treated as a kind-of Form trait, where multiple fields could be treated out of the context.

In this case, we may just ignore it, and refactor it later - it may be too complex to fix it now.

Reply:
Marking as resolved then, and we will just handle it later 👍